File: mobile/src/main/java/com/windscribe/mobile/welcome/WelcomePresenter.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.welcome;

import android.net.Uri;

public interface WelcomePresenter {

    void exportLog();

    boolean isUserPro();

    void onBackPressed();

    void onDestroy();

    void printStartUri(Uri uri);

    void startAccountClaim(String username, String password, String email, boolean ignoreEmptyEmail);

    void startGhostAccountSetup();

    void startLoginProcess(String username, String password, String twoFa);

    void startSignUpProcess(String username, String password, String email, boolean ignoreEmptyEmail);

}

File: mobile/src/main/java/com/windscribe/mobile/welcome/WelcomePresenter.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.welcome

import java.io.File

interface WelcomePresenter {
    fun exportLog()
    val isUserPro: Boolean
    fun onBackPressed()
    fun onDestroy()
    fun startAccountClaim(
        username: String,
        password: String,
        email: String,
        ignoreEmptyEmail: Boolean,
        voucherCode: String
    )

    fun startGhostAccountSetup()
    fun startLoginProcess(username: String, password: String, twoFa: String)
    fun startSignUpProcess(
        username: String,
        password: String,
        email: String,
        referralUsername: String,
        ignoreEmptyEmail: Boolean,
        voucherCode: String
    )

    fun getLogUri(): File?
}